Src/FluentAssertions/Equivalency/Steps/GenericDictionaryEquivalencyStep.cs

```diff
@@ -175,32 +156,39 @@ private static bool AssertSubjectIsNotNull(object subject)
         IDictionary<TExpectedKey, TExpectedValue> expectation)
         where TExpectedKey : TSubjectKey
     {
-        foreach (TExpectedKey key in expectation.Keys)
+        if (subject.Count != expectation.Count)
+        {
+            AssertSameLength(subject, expectation);
```

Comment:
⛏️ In a way, we've already asserted that the length isn't the same. Now we just want to report what's missing. Maybe we should capture that in the name.